open-banking-accelerator/components/com.wso2.openbanking.accelerator.gateway/src/main/java/com/wso2/openbanking/accelerator/gateway/executor/dcr/DCRExecutor.java

```diff
@@ -610,6 +250,409 @@ public void postProcessRequest(OBAPIRequestContext obapiRequestContext) {
+                    apiIDList = filterRegulatoryAPIs(regulatoryAPIs, publishedAPIsResponse.getAsJsonObject()
+                            .get(OpenBankingConstants.API_LIST).getAsJsonArray(), getRolesFromSSA(softwareStatement));
+                }
+            } else {
+                log.warn("No regulatory APIs configured. Application will be subscribed to all published APIs");
+                //subscribe to all APIs if there are no configured regulatory APIs
+                for (JsonElement apiInfo : publishedAPIsResponse.getAsJsonObject().get("list").getAsJsonArray()) {
+                    apiIDList.add(apiInfo.getAsJsonObject().get("id").getAsString());
+                }
+            }
+            //subscribe to apis
+            JsonArray subscribeAPIsPayload = getAPISubscriptionPayload(amApplicationCreateResponse
+                    .getAsJsonObject().get(applicationIdParam).getAsString(), apiIDList);
+            JsonElement subscribeAPIsResponse = callPost(urlMap.get(GatewayConstants.API_SUBSCRIBE_URL).toString(),
+                    subscribeAPIsPayload.toString(), "Bearer ".concat(token));
+            if (subscribeAPIsResponse == null) {
+                log.error("Error while subscribing to APIs");
+                String clientId = createdDCRAppDetails.get(clientIdParam).getAsString();
+                //delete service provider
+                callDelete(fullBackEndURL.concat("/").concat(clientId), basicAuthHeader);
+                //delete SP created for calling dev portal REST APIs
+                callDelete(urlMap.get(GatewayConstants.IAM_DCR_URL).toString().concat("/")
+                        .concat(amRestAPIInvokeClientId), basicAuthHeader);
+                //delete AM application
+                callDelete(urlMap.get(GatewayConstants.APP_CREATE_URL).toString()
+                        .concat("/").concat(amApplicationCreateResponse.getAsJsonObject()
+                                .get(applicationIdParam).getAsString()), GatewayConstants.BEARER_TAG.concat(token));
+                handleInternalServerError(obapiResponseContext, OpenBankingErrorCodes.REGISTRATION_INTERNAL_ERROR);
+                return;
+            }
+
+            //delete IAM application used to invoke am rest endpoints
+            if (!callDelete(urlMap.get(GatewayConstants.IAM_DCR_URL).toString().concat("/")
+                    .concat(amRestAPIInvokeClientId), basicAuthHeader)) {
+                handleInternalServerError(obapiResponseContext, OpenBankingErrorCodes.REGISTATION_DELETE_ERROR);
+            }
+
+        } catch (IOException | OpenBankingException | URISyntaxException | ParseException e) {
+            log.error("Error occurred while creating application", e);
+            handleInternalServerError(obapiResponseContext, OpenBankingErrorCodes.REGISTRATION_INTERNAL_ERROR);
+        }
+    }
+
+    private void processPutRequestResponse(OBAPIResponseContext obapiResponseContext, String basicAuthHeader,
```

Comment:
Add method comment

Reply:
Resolved in [8657c32](https://github.com/wso2/financial-open-banking/pull/62/commits/8657c32a1adada4bcf5b652287b825b09a36ca96)